dashboard/lib/services/user/pii_scrubber.rb

```diff
@@ -0,0 +1,118 @@
+        user.reset_password_token = nil
+        user.unlock_token = nil
+        user.oauth_refresh_token = nil
+        user.oauth_token = nil
+        user.oauth_token_expiration = nil
+        user.parent_email = nil
+        user.secret_picture_id = nil
+        user.secret_words = nil
+        user.studio_person&.destroy # Studio person record may contain emails
+
+        # Users might have multiple accounts with the same email address.
+        # If there is a live user with the same email, these data points will not be scrubbed.
+        if email.present? && ::User.find_by_email(email).blank?
+          EmailPreference.where(email: email).destroy_all
+          census_submissions = Census::CensusSubmission.where(submitter_email_address: email)
+          csfms = Census::CensusSubmissionFormMap.where(census_submission_id: census_submissions.pluck(:id))
+          csfms.destroy_all
+          census_submissions.destroy_all
+        end
+
+        # Names
+        user.name = nil
+        user.family_name = nil
+        user.username = SecureRandom.alphanumeric(20)
+        user.ops_first_name = nil
+        user.ops_last_name = nil
+
+        # Address and location data
+        user.full_address = nil
+        user.current_sign_in_ip = nil
+        user.last_sign_in_ip = nil
+        user.data_transfer_agreement_request_ip = nil
+        user.user_geos.each(&:clear_user_geo)
+      end
+
+      # Legacy delete acccounts helper client for purging data from deprecated tables
+      # and Pegasus DB. After Pegasus DB is fully deprecated, we should remove this dependency.
+      private def delete_accounts_helper
+        @delete_accounts_helper ||= DeleteAccountsHelper.new(bypass_safety_constraints: true)
+      end
+
+      # Deletes PII from deprecated tables that no longer have a corresponding ActiveRecord model.
+      # Also deletes PII from Pegasus DB, which is planned to for deprecation and should not generally
+      # be accessed from Dashboard code. Pegasus DB is a separate database in the RDS cluster, and includes
+      # PII such as:
+      # - PD applications and forms with email addresses, names, addresses
+      # - Email addresses in Poste tables
+      # - Location and email data in "forms" tables
+      # - Email addresses in contact rollup tables
+      private def scrub_legacy_data
+        if email.present?
+          delete_accounts_helper.clean_and_destroy_pd_content(user.id, email)
+          delete_accounts_helper.remove_poste_data(email)
+          delete_accounts_helper.remove_census_submissions(email)
+          delete_accounts_helper.purge_contact_rollups(email)
+        end
+        delete_accounts_helper.clean_pegasus_forms_for_user(user)
+      end
+
+      # Removes any third-party data that requires an API call. Called after
+      # other methods since it is not reversible.
+      private def scrub_external_data
+        MailJet.delete_contact(email) if email.present? && ::User.find_by_email(email).blank?
```

Comment:
1. `User.exists?` => `SELECT 1 AS one FROM users WHERE users.deleted_at IS NULL AND users.email = 'expected_email' LIMIT 1`
2. `User.exists?(email:)` =>  Starting Ruby v3.1 we can define a hash just like `{email:}` instead of `{email: email}`
```suggestion
        MailJet.delete_contact(email) if email.present? && !::User.exists?(email: email)
```